packages/fluentui/react-bindings/src/styles/resolveVariables.ts

```diff
@@ -10,33 +10,50 @@ import {
 const variablesCache = new WeakMap<ThemePrepared, Record<string, ComponentVariablesObject>>();
 
 const resolveVariables = (
-  displayName: string,
+  displayNames: string[],
   theme: ThemePrepared,
   variables: ComponentVariablesInput | undefined,
   enabledVariablesCaching: boolean | undefined
 ): ComponentVariablesObject => {
+  let componentThemeVariables: ComponentVariablesObject;
+
+  // Filter out components that doesn't have defined variables in theme
+  const effectiveDisplayNames = displayNames.filter(displayName => !!theme.componentVariables[displayName]);
+
   //
   // Simple caching model, works only if there is no `props.variables`
   // Resolves variables for this component, cache the result in provider
   //
 
-  let componentThemeVariables = {};
-
   if (enabledVariablesCaching) {
     if (!variablesCache.has(theme)) {
       variablesCache.set(theme, {});
     }
 
     const variablesThemeCache = variablesCache.get(theme) || {};
 
-    if (!variablesThemeCache[displayName]) {
-      variablesThemeCache[displayName] = callable(theme.componentVariables[displayName])(theme.siteVariables) || {};
+    // const displayNames = { Foo: variables, Bar: undefined, Baz: undefined }
+    // This allows to avoid creating useless cache entries for `Bar` & `Baz` components
+    const handlingDisplayName = effectiveDisplayNames[effectiveDisplayNames.length - 1];
+
+    if (!variablesThemeCache[handlingDisplayName]) {
+      effectiveDisplayNames.map(displayName => theme.componentVariables[displayName]);
+
+      variablesThemeCache[handlingDisplayName] = mergeComponentVariables(
+        ...effectiveDisplayNames.map(displayName => theme.componentVariables[displayName])
+      )(theme.siteVariables);
```

Comment:
May be a premature optimization, but maybe we can reuse here prev caching if we find for some other displayName..

Reply:
Can you please add a code snippet? Not sure that I get your idea 🤔

Reply:
NVM, we need to merge those, we cannot reuse prev results.